system/jlib/jmutex.hpp

```diff
@@ -288,7 +288,15 @@ class CriticalSection
     inline ~CriticalSection()
     {
 #ifdef _ASSERT_LOCK_SUPPORT
-        assertex(owner==0 && depth==0);
```

Comment:
This feels like it might be adding some overhead to a very commonly used bit of code...